SDK is released when tag is created

shinsuke-nara opened the pull request:
JSSDKでは、既にdeploymentセクションでリリースされるようになっていました。

他のSDKと揃えて、タグが作成された時点でリリースが行われるよう修正しました。

Related issue: No. 1021

satoshikumano commented:
@yongpingchen can you confirm?

yongpingchen commented on circle.yml:
```diff
@@ -18,7 +18,7 @@ test:
 deployment:
   npm:
-    branch: master
+    tag: /[0-9]+.*/
```
should not match non number in the middle expect `.`, and tag should end with number.

shinsuke-nara replied:
@yongpingchen 

I agree with kumano-san that using non number at the end of tag name is acceptable at to this [comment](https://github.com/KiiCorp/AndroidStorageSDK/pull/1946#discussion_r89045343). I expect `0beta2` can be used as a tag name.

According to the comment, I adopt the regular expression.

Could you talk about this with kumano-san? I follow the decision.

yongpingchen replied:
After discussed with Kumano-san, the regx can be `[0-9]+(\.[0-9]+){0,2}(-.*)?`(but please check carefully). 

### Expected match
1
0.3
0.4.5
0.4.5-beta
0.3-beta

### Expected non-match
0beta
0.1.1.1 (more than 2 dots)
abc
1.abc

shinsuke-nara commented:
@yongpingchen 

I fixed regular expression to match tag name according to your comment.
I checked your suggested regular expression. The regular expression is not matched tag name which you expect and matched tag name which you do not expect.

So I fixed it. Could you check this?

yongpingchen commented:
@shinsuke-nara 
The expression mismatched one possible like `1-beta`. 
So I think it should be `^[0-9]+(\.[0-9]+){0,2}((-.*)?)?$`
What do you think?

shinsuke-nara commented:
@yongpingchen 
I fixed according to your comment. Please check this.

yongpingchen commented:
LGTM
